src/esm_runscripts/filedicts.py

```diff
@@ -314,10 +317,39 @@ def _resolve_paths(self) -> None:
         for key, path in self.locations.items():
             self[f"absolute_path_in_{key}"] = path.joinpath(self[f"name_in_{key}"])
 
-    def _path_type(self, path: pathlib.Path) -> Union[str, bool]:
+    def convert_to_path(self, path: Union[str, pathlib.Path]) -> pathlib.Path:
```

Comment:
I removed the method `convert_to_path` and the test associated with that but now I am getting failing tests because we can still have non `str` or `Path` object and there are no checks covering that.